### PR TITLE
Recent Feed Overhaul: Add engagement bar

### PR DESCRIPTION
**Pull request by DustyReagan**
Related to https://github.com/Automattic/loop/issues/195

## Proposed Changes

* This PR reuses the `ReaderPostActions` component to create a new `EngagementBar` that we use on the `Recent` posts view. The `EngagementBar` contains the same features as the "Post's Actions."
* The `EngagementBar` is fixed at the bottom of the post screen and visible when the "Post Actions" are off-screen. When the "Post Actions" are in the viewport, the `EngagementBar` is hidden.

Demo

https://github.com/user-attachments/assets/57c589e8-7b54-4cd9-b343-293086e7a122

## Why are these changes being made?

* To encourage user engagement with the post.

## Testing Instructions

* Use Calypso Live and go to /read?flags=reader/recent-feed-overhaul
* Interact with the engagement bar in various posts and test that it is working. Try to break it.
* Check the engagement bar in various widths and mobile view.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by eoigal** [comment]
Small glitch I think on the share popup (I think it should disappear on scroll?)


https://github.com/user-attachments/assets/0f59cb77-c68d-4610-8829-7bcc3beda445

**Comment by davemart-in** [comment]
A couple of minor things:

1. When you click comment, can we add focus to the comment field, so people can just start typing?
2. On mobile, can we add a bit more padding between each of the icons:

CleanShot 2024-11-11 at 07 40 39@2x

File: client/blocks/reader-full-post/index.jsx [hunk 1]
```diff
@@ -286,41 +286,21 @@ export class FullPostView extends Component {
...
-						// check to see if the comment node moved while we were scrolling
-						// and scroll to the end position
-						const commentsNodeAfterScroll = this.commentsWrapper.current;
-						if ( commentsNodeAfterScroll && commentsNodeAfterScroll.offsetTop ) {
-							window.scrollTo( 0, commentsNodeAfterScroll.offsetTop - 48 );
-						}
-						this._scrolling = false;
-					},
-				} );
+		scrollToComments( {
```
Comment (DustyReagan) [inline client/blocks/reader-full-post/index.jsx:295]: I moved this to a new file client/blocks/reader-full-post/scroll-to-comments.tsx so I could reuse the logic inside the EngagmentBar component [here](https://github.com/Automattic/wp-calypso/pull/96172/files#diff-dcc9916e9c34067da838dee64179aaf8aca0c92026d439fc097b02f928834173R28).

**Comment by DustyReagan** [comment]
Thanks for the review, @davemart-in and @eoigal! I've addressed your feedback and updated this PR. It's ready for another look. 🙏

**Comment by eoigal** [comment]
Still seeing the share and repost popups when the engagement bar disappears - not 100% if this is an issue though!

https://github.com/user-attachments/assets/1521452e-f8cf-4d04-8602-fb947c320b88

**Review by davemart-in — APPROVED** [review]
Looks good enough to ship behind a feature flag!

**Review by eoigal — APPROVED** [review]
WFM! Not sure if anything changed but popups are disappearing as expected now.

**Pull request merged**